db.py

```diff
@@ -42,34 +41,56 @@ def addView(self, connection, url):
 
     def getTopSites(self, connection, amount=10):
         """ Get the top domains using this tool by hits. Ignore specified domains """
+        top_urls = self.getTopUrls(connection, -1)
+
+        # Get total hits per domain
+        site_counts = defaultdict(int)
+        for url in top_urls['urls']:
+            # Get the domain - part before the first '/'
+            domain = url.split('/')[0]
+            # Add hit counts to the domain
+            site_counts[domain] += top_urls['values'][url]
+
+        # Sort the domains by hits
+        sorted_sites = sorted(site_counts, key=lambda x: site_counts[x], reverse=True)
+
+        # Return sorted domains and their values, this allows for lower Python version support
+        return {
+            'domains': sorted_sites[:amount],
+            'values': {site: site_counts[site] for site in site_counts}
+        }
+
+    def getTopUrls(self, connection, amount=10):
+        """ Get the top urls using this tool by hits. Ignore specified domains """
         # Select all urls and counts
+        if amount < 0:
+            amount = 2147483647
         cursor = connection.cursor()
-        cursor.execute('select url, count from url')
+        cursor.execute(f'select url, count from url order by count desc limit {amount + len(config.TOP_SITES_IGNORE_DOMAIN_RE_MATCH)}')
```

Comment:
If `amount` is 10 and `config.TOP_SITES_IGNORE_DOMAIN_RE_MATCH` is empty, only 10 URLs will be selected. Also what is `config.TOP_SITES_IGNORE_DOMAIN_RE_MATCH` needed for here?

Reply:
**TL;DR:** Didn't make sense, sorry. I'm simply gonna revert this change.

My thoughts were:
* Passing `amount=10`, you actually want to end up with 10 entries, in case there are at least 10 entries available, that don't match the ignore query
* If you do `LIMIT 10`, but two of the results are getting ignored, the user will only end up with 8 results, although there might have been more matches in the database, that would not have been ignored
* I thought: "let's just return as many items more, as there are ignore-patterns", but that didn't make any sense, as every pattern can match an arbitrary number of result entries